src/quacc/utils/dicts.py

```diff
@@ -95,7 +97,7 @@ def _recursive_dict_pair_merge(
     """
     dict1 = dict1 or ({} if dict1 is None else dict1.__class__())
     dict2 = dict2 or ({} if dict2 is None else dict2.__class__())
-    merged = safe_dict_copy(dict1)
+    merged = safe_dict_copy(dict(dict1))
```

Comment:
I guess this and the change on line 184 are needed because `safe_dict_copy` and `sort_dict` are only type hinted for `dict`. Should we be type hinting them for `MutableMapping` instead?

Reply:
Yes, that makes sense. I updated all type hints from `dict` to `MutableMapping` in `dicts.py`